lib/ssh_scan/worker.rb

```diff
@@ -0,0 +1,62 @@
+require 'ssh_scan/scan_engine'
+require 'openssl'
+require 'net/https'
+
+module SSHScan
+  class Worker
+    def initialize(opts = {})
+      @server = opts[:server] || "127.0.0.1"
+      @port = opts[:port] || 8000
+      @logger = opts[:logger] || Logger.new(STDOUT)
+      @poll_interval = 5 # seconds
+      @worker_id = SecureRandom.uuid
+      @verify_ssl = false
+    end
+
+    def get_work
+      (Net::HTTP::SSL_IVNAMES << :@ssl_options).uniq!
+      (Net::HTTP::SSL_ATTRIBUTES << :options).uniq!
+
+      Net::HTTP.class_eval do
+        attr_accessor :ssl_options
+      end
+
+      uri = URI("https://#{@server}:#{@port}/api/v0.0.1/work?worker_id=#{@worker_id}")
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      http.verify_mode = OpenSSL::SSL::VERIFY_NONE # unless @verify_ssl == true
+      options_mask = OpenSSL::SSL::OP_NO_SSLv2 + OpenSSL::SSL::OP_NO_SSLv3 + OpenSSL::SSL::OP_NO_COMPRESSION
+      http.ssl_options = options_mask
+      request = Net::HTTP::Get.new(uri.path)
+      response = http.request(request)
+      JSON.parse(response.body)
```

Comment:
This is missing a lot of exception handling and will just die if it runs into a bad HTTP response.